### ISSUE TITLE
Use MPI_Exscan to compute offsets?

### ISSUE THREAD

**Issue by adammoody** [ECP-VeloC/VELOC#42]
Just browsing through code, I noticed this pattern:

https://github.com/ECP-VeloC/VELOC/blob/c8576889ffda81a8200b916a04f63d5380d6282d/src/lib/client.cpp#L194-L199

I suspect you might be able to replace that code segment with an ``MPI_Scan`` or ``MPI_Exscan``, which takes O(log P) time instead of O(P).

**Comment by bnicolae** [comment]
Indeed, the prefix sum is in the critical checkpointing path that is blocking the app. Switched to MPI_Exsend.
